scripts/updateContributors.mjs

```diff
@@ -0,0 +1,76 @@
+import simpleGit from 'simple-git';
+import { Octokit } from '@octokit/rest';
+import fs from 'node:fs';
+import path from 'path';
+import pMemoize from 'p-memoize';
+
+const IGNORED_COMMITS = ['433bbae4f1d4abb50a26306d6679a38ace5c8b78'];
+
+const octokit = new Octokit({ auth: process.env.GITHUB_TOKEN });
+
+const cache = new Map();
+
+const getUserName = pMemoize(
+  async (commit) => {
+    const { data: fetchedCommit } = await octokit.repos.getCommit({
+      owner: 'lucide-icons',
+      repo: 'lucide',
+      ref: commit.hash,
+    });
+    return fetchedCommit?.author?.login;
+  },
+  { cacheKey: ([commit]) => commit.author_email, cache }
+);
+
+const getContributors = async (file, includeCoAuthors) => {
+  const { all: commits } = await simpleGit().log(['--reverse', '--follow', '--', file]);
+
+  const emails = new Map();
+  for (let i = 0; i < commits.length; i += 1) {
+    const commit = commits[i];
+    if (!IGNORED_COMMITS.includes(commit.hash)) {
+      if (!emails.has(commit.author_email)) {
+        emails.set(commit.author_email, getUserName(commit));
+      }
+      if (includeCoAuthors) {
+        const matches = commit.body.matchAll(
+          /(^Author:|^Co-authored-by:)\s+(?<author>[^<]+)\s+<(?<email>[^>]+)>/gm
+        );
+        // eslint-disable-next-line no-restricted-syntax
+        for (const match of matches) {
+          if (!emails.has(match.groups.email) && cache.has(match.groups.email)) {
+            emails.set(match.groups.email, Promise.resolve(cache.get(match.groups.email)));
+          }
+        }
+      }
+    }
+  }
+
+  return Promise.all(Array.from(emails.values()));
+};
+
+const files = process.argv
+  .slice(2)
+  .map((file) => file.replace('.json', '.svg'))
+  .filter((file, idx, arr) => arr.indexOf(file) === idx);
+
+if (!files.every((file) => file.startsWith('icons/'))) {
+  console.error("file path must start with 'icons/'");
+  process.exit(1);
+}
+
+// get all contributors to be able to associate them when they are co authors
+await getContributors('icons');
+
+await Promise.all(
+  files.map(async (file) => {
+    const jsonFile = path.join(process.cwd(), file.replace('.svg', '.json'));
+    const json = JSON.parse(fs.readFileSync(jsonFile));
+    const contributors = [
+      ...(json.contributors || []),
+      ...(await getContributors(file, true)),
+    ].filter((contributor, idx, arr) => contributor && arr.indexOf(contributor) === idx);
+
+    fs.writeFileSync(jsonFile, JSON.stringify({ ...json, contributors }, null, 2));
```

Comment:
Maybe it's better if we change the order of the object properties to:
```json
{
  "$schema": "..."
  "contributors": [],
  "tags": [],
  "categories": []
}
```
This way I think we prevent the issue with a possible merge conflict.

Reply:
@ericfennis: I've rebased and added this reordering (as well as updated contributors to be up to date with latest changes as well as the reordering).

Reply:
And again. 😅